src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1200 @@
+            if (LOG.isInfoEnabled()) {
+                LOG.info("Saved Advance Deposit document: " + document.getDocumentNumber());
+            }
+        } catch (Exception e) {
+            LOG.error("Error persisting document # " + document.getDocumentHeader().getDocumentNumber() + " " + e.getMessage(), e);
+            throw new RuntimeException("Error persisting document # " + document.getDocumentHeader().getDocumentNumber() + " " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * @see AdvanceDepositService#routeAdvanceDepositDocuments()
+     */
+    public boolean routeAdvanceDepositDocuments() {
+        List<String> documentIdList;
+
+        try {
+            documentIdList = retrieveAdvanceDepositDocumentsToRoute(KewApiConstants.ROUTE_HEADER_SAVED_CD);
+        } catch (WorkflowException | RemoteException e) {
+            LOG.error("Error retrieving advance deposit documents for routing: " + e.getMessage(), e);
+            throw new RuntimeException(e.getMessage(), e);
+        }
+
+        if (LOG.isInfoEnabled()) {
+            LOG.info("Advance deposit to Route: " + documentIdList);
+        }
+
+        for (String documentId : documentIdList) {
+            try {
+                AdvanceDepositDocument advanceDocument = (AdvanceDepositDocument) documentService.getByDocumentHeaderId(documentId);
+                if (LOG.isInfoEnabled()) {
+                    LOG.info("Routing advance deposit document # " + documentId + ".");
+                }
+                documentService.prepareWorkflowDocument(advanceDocument);
+                documentService.routeDocument(advanceDocument, "document routed by achIncome batch job", null);
+            } catch (WorkflowException e) {
+                LOG.error("Error routing document # " + documentId + " " + e.getMessage());
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns a list of all initiated but not yet routed advance deposit documents, using the WorkflowDocumentService.
+     *
+     * @return a list of advance deposit documents to route
+     */
+    protected List<String> retrieveAdvanceDepositDocumentsToRoute(String statusCode) throws WorkflowException, RemoteException {
+        List<String> documentIds = new ArrayList<String>();
+
+        List<DocumentStatus> routeStatuses = new ArrayList<DocumentStatus>();
+        routeStatuses.add(DocumentStatus.fromCode(statusCode));
+
+        Person systemUser = getPersonService().getPersonByPrincipalName(KFSConstants.SYSTEM_USER);
+        String principalName = systemUser.getPrincipalName();
+
+        DocumentSearchCriteria.Builder criteria = DocumentSearchCriteria.Builder.create();
+        criteria.setDocumentTypeName(KFSConstants.FinancialDocumentTypeCodes.ADVANCE_DEPOSIT);
+        criteria.setDocumentStatuses(routeStatuses);
+        criteria.setInitiatorPrincipalName(principalName);
```

Comment:
I checked with Cathy and also Kymber from the product team and they didn't know of any. Also, FWIW, I had to explicitly give the system user permissions to route the AD documents, so if there was something it would just be saving the documents and not routing them which seems odd.